tuxemon/save.py

```diff
@@ -52,127 +54,256 @@
 
 slot_number = None
 TIME_FORMAT = "%Y-%m-%d %H:%M"
-
-
-def get_save_data(session: Session) -> Mapping[str, Any]:
-    """
-    Gets a dictionary which represents the state of the session.
-
-    Parameters:
-        session: Game session.
-
-    Returns:
-        Game data to save, must be JSON encodable.
-
-    """
-    save_data = session.player.get_state(session)
-    screenshot = capture_screenshot(session.client)
-    save_data["screenshot"] = base64.b64encode(pygame.image.tostring(screenshot, "RGB")).decode("utf-8")
-    save_data["screenshot_width"] = screenshot.get_width()
-    save_data["screenshot_height"] = screenshot.get_height()
-    save_data["time"] = datetime.datetime.now().strftime(TIME_FORMAT)
-    save_data["version"] = SAVE_VERSION
-    return save_data
-
-
-def capture_screenshot(client: LocalPygameClient) -> pygame.surface.Surface:
-    """
-    Capture a screenshot.
-
-    Parameters:
-        client: Tuxemon client.
-
-    Returns:
-        Captured image.
-
-    """
-    screenshot = pygame.Surface(client.screen.get_size())
-    world = client.get_state_by_name(WorldState)
-    assert world
-    world.draw(screenshot)
-    return screenshot
-
-
-def save(
-    save_data: Mapping[str, Any],
-    slot: int,
-) -> None:
-    """
-    Saves the current game state to a file using CBOR or JSON.
-
-    Parameters:
-        save_data: The data to save.
-        slot: The save slot to save the data to.
-
-    """
-    # Save a screenshot of the current frame
-    save_path = prepare.SAVE_PATH + str(slot) + ".save"
-    if prepare.SAVE_METHOD == "CBOR":
-        text = cbor.dumps(save_data)
-    else:
-        text = json.dumps(save_data, indent=4, separators=(",", ": "))
-    with open(save_path, "w") as f:
-        logger.info("Saving data to save file: " + save_path)
-        # Don't dump straight to the file: if we crash it would corrupt
-        # the save_data
-        f.write(text)
-
-
-def load(slot: int) -> Optional[Mapping[str, Any]]:
-    """
-    Loads game state data from a save file.
-
-    Parameters:
-        slot: The save slot to load game data from.
-
-    Returns:
-        Dictionary containing game data to load.
-
-    """
-    save_path = f"{prepare.SAVE_PATH}{slot}.save"
-    save_data = open_save_file(save_path)
-    if save_data:
-        return upgrade_save(save_data)
-    elif save_data is None:
-        # File not found; it probably wasn't ever created, so don't panic
-        return None
-    else:
-        save_data["error"] = "Save file corrupted"
-        save_data["player_name"] = "BROKEN SAVE!"
-        logger.error("Failed loading save file.")
+config = prepare.CONFIG
+
+if config.compress_save:
```

Comment:
why is all this code duplicated?  i would think that the original json saves could be used (pass them into the compress method?) instead of duplicating the code?